core/src/main/java/org/infinispan/commands/functional/Mutations.java

```diff
@@ -89,15 +122,21 @@ public byte type() {
 
       @Override
       public R apply(EntryView.ReadWriteEntryView<K, V> view) {
-         return f.apply(value, view);
+         return f.apply((V) valueDataConversion.fromStorage(value), view);
+      }
+
+      @Override
+      public void inject(ComponentRegistry registry) {
+         registry.wireDependencies(valueDataConversion);
```

Comment:
I only used `wireDependency` for `CacheTopologyControlCommand` because I knew those are pretty rare, I don't think it's efficient enough to use multiple times per operation.